include/flang/Parser/source.h

```diff
@@ -44,16 +47,16 @@ class SourceFile {
   std::size_t lines() const { return lineStart_.size(); }
   Encoding encoding() const { return encoding_; }
 
-  bool Open(std::string path, std::stringstream *error);
-  bool ReadStandardInput(std::stringstream *error);
+  bool Open(std::string path, llvm::raw_ostream *error);
+  bool ReadStandardInput(llvm::raw_ostream *error);
```

Comment:
Why is this not raw_string_ostream?

Reply:
There's no need for these functions to take the more specific type really, we may for example eventually want to call or test them with a different stream type.

They should be references rather than pointers though now that you point it out

Reply:
OK. If that is the case then please keep it consistent or provide justification on why raw_string_stream is used in functions Open and ReadStandardInput in Parser/provenance.cpp and TokenSequence::Put in Parser/token-sequence.cpp